cpp/src/arrow/util/CMakeLists.txt

```diff
@@ -33,15 +33,17 @@ install(FILES
   compression_zstd.h
   cpu-info.h
   decimal.h
-  hash-util.h
   hash.h
+  hash-util.h
+  io-util.h
```

Comment:
Looks like `memory.h` is also missing

Reply:
Can we just glob files to avoid having to update this list?

Reply:
Yeah, that would be better, taking a look